packages/api-explorer/src/components/SelectorContainer/SdkLanguageSelector.tsx

```diff
@@ -55,6 +57,20 @@ export const SdkLanguageSelector: FC = () => {
     setSdkLanguageAction({ sdkLanguage: language })
   }
 
+  useEffect(() => {
+    const searchParams = new URLSearchParams(history.location.search)
+    const urlSdk = searchParams.get('sdk')
+    if (urlSdk && allSdkLanguages.find((lang) => lang.label === urlSdk)) {
+      setSdkLanguageAction({ sdkLanguage: urlSdk! })
+    }
+  }, [])
```

Comment:
if `APIExplorer.tsx` is handling reading query params and driving state accordingly shouldn't this entire effect be redundant?

Reply:
Agree, I had a lapse of judgement here.